src/FileWriterTask.cpp

```diff
@@ -25,119 +25,120 @@ json hdf_parse(std::string const &Structure) {
+  }
 }
 
-void FileWriterTask::add_source(Source &&source) {
-  if (UseHDFSWMR) {
-    source.HDFFileForSWMR = &hdf_file;
+void FileWriterTask::addSource(Source &&Source) {
+  if (swmrEnabled()) {
+    Source.HDFFileForSWMR = &hdfFile;
   }
-  bool found = false;
-  for (auto &d : _demuxers) {
-    if (d.topic() == source.topic()) {
-      d.add_source(std::move(source));
-      found = true;
+
+  // If source already exists then replace
+  for (auto &d : Demuxers) {
+    if (d.topic() == Source.topic()) {
+      d.add_source(std::move(Source));
+      return;
     }
   }
-  if (!found) {
-    _demuxers.emplace_back(source.topic());
-    auto &d = _demuxers.back();
-    d.add_source(std::move(source));
-  }
-}
 
-void FileWriterTask::hdf_init(std::string const &NexusStructure,
-                              std::string const &ConfigFile,
-                              std::vector<StreamHDFInfo> &stream_hdf_info) {
-  filename_full = hdf_filename;
-  if (!hdf_output_prefix.empty()) {
-    filename_full = hdf_output_prefix + "/" + filename_full;
-  }
+  // Add new source
+  Demuxers.emplace_back(Source.topic());
+  auto &d = Demuxers.back();
+  d.add_source(std::move(Source));
+}
 
+void FileWriterTask::InitialiseHdf(std::string const &NexusStructure,
+                                   std::string const &ConfigFile,
+                                   std::vector<StreamHDFInfo> &HdfInfo,
+                                   bool UseSwmr) {
   auto NexusStructureJson = hdf_parse(NexusStructure);
   auto ConfigFileJson = hdf_parse(ConfigFile);
 
   try {
-    LOG(Sev::Info, "Creating HDF file {}", filename_full);
-    hdf_file.init(filename_full, NexusStructureJson, ConfigFileJson,
-                  stream_hdf_info, UseHDFSWMR);
-  } catch (...) {
-    std::throw_with_nested(std::runtime_error(fmt::format(
-        "can not initialize hdf file  hdf_output_prefix: {}  hdf_filename: {}",
-        hdf_output_prefix, hdf_filename)));
+    LOG(Sev::Info, "Creating HDF file {}", Filename);
+    hdfFile.init(Filename, NexusStructureJson, ConfigFileJson, HdfInfo,
+                 UseSwmr);
+    // The HDF file is closed and re-opened to (optionally) support SWMR and
+    // parallel writing.
+    closeFile();
+    reopenFile();
+
+  } catch (std::exception const &E) {
+    std::throw_with_nested(std::runtime_error(
+        fmt::format("can not initialize hdf file {}", Filename)));
   }
 }
 
-void FileWriterTask::hdf_close_before_reopen() { hdf_file.close(); }
+void FileWriterTask::closeFile() { hdfFile.close(); }
 
-int FileWriterTask::hdf_reopen() {
+void FileWriterTask::reopenFile() {
   try {
-    hdf_file.reopen(filename_full, json::object());
+    hdfFile.reopen(Filename, json::object());
   } catch (std::exception const &E) {
     LOG(Sev::Error, "Exception: {}", E.what());
     if (StatusProducer) {
-      logEvent(StatusProducer, StatusCode::Error, ServiceID, JobID,
+      logEvent(StatusProducer, StatusCode::Error, ServiceId, JobId,
```

Comment:
Is there much point having any of these log comments here if you're just going to rethrow the exception?

As well as this, is there any benefit to catching and then rethrowing the exception, it may as well be left uncaught.